contracts/refs/OracleRef.sol

```diff
@@ -92,6 +92,8 @@ abstract contract OracleRef is IOracleRef, CoreRef {
 
         // Scale the oracle price by token decimals delta if necessary
         uint256 scalingFactor;
+
+        // TODO: Broken abstraction, this needs pushing down
```

Comment:
I like being able to invert & normalize decimals at this level, but the fact that they can be set after deployment (setDoInvert and setDecimalsNormalizer) is weird, because for a given deployment that should be fixed. These functions are governor-only, so I guess that's fine.

Reply:
Let's discuss this oracle topic at the next dev meeting

Reply:
Why do you want to be able to invert and normalise at this level? That's an oracle specific implementation detail - how those factors are taken into account differs for each oracle. 

For example, I believe with the Chainlink oracle it's a simple inversion and multiplication once the price is retrieved. However, for the Uniswap TWAP oracle it requires mixing these factors into the intermediate numerator and denominator that you calculate

Reply:
Refactoring `OracleRef.sol` is a significant effort and it's unlikely we will call `setDoInvert()` and `setDecimalsNormalizer()`, as you say governor only. But it feels very dangerous to set a `backupOracle` which will report a wildly incorrect price if those methods are ever accidentally called.